airbyte-integrations/connectors/source-github/source_github/availability_strategies.py

```diff
@@ -0,0 +1,88 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+import logging
+from typing import Dict, Optional
+
+import requests
+from airbyte_cdk.sources import Source
+from airbyte_cdk.sources.streams.core import Stream
+from airbyte_cdk.sources.streams.http.availability_strategy import HttpAvailabilityStrategy
+from airbyte_cdk.sources.utils.stream_helpers import StreamHelper
+from requests import HTTPError
+
+
+class OrganizationBasedAvailabilityStrategy(HttpAvailabilityStrategy):
+    """
+    Availability Strategy for organization-based streams.
+    """
+
+    def reasons_for_unavailable_status_codes(
+        self, stream: Stream, logger: logging.Logger, source: Optional["Source"], error: HTTPError
+    ) -> Dict[int, str]:
+        stream_slice = StreamHelper().get_stream_slice(stream)
+        organisation = stream_slice.get("organization", "")
```

Comment:
I new implementation we can be more strict now
```
stream_slice["organization"]
```

instead of `stream_slice.get("organization", "")`